tools/cli/commands.go

```diff
@@ -205,29 +205,27 @@ var workflowClosedStatusMap = map[string]s.WorkflowExecutionCloseStatus{
 // To provide customized builder, call SetBuilder() before call NewCliApp()
 var cBuilder WorkflowClientBuilderInterface
 
+// osExit is used when CLI hits an error and exit
+// The purpose of this is to test CLI exit scenario
+var osExit = os.Exit
+
 // SetBuilder can be used to inject customized builder of cadence clients
 func SetBuilder(builder WorkflowClientBuilderInterface) {
 	cBuilder = builder
 }
 
 // ErrorAndExit print easy to understand error msg first then error detail in a new line
 func ErrorAndExit(msg string, err error) {
-	fmt.Printf("%s %s\n%s %+v\n", colorRed("Error:"), msg, colorMagenta("Error Details:"), err)
-	os.Exit(1)
-}
-
-// ExitIfError exit while err is not nil and print the calling stack also
-func ExitIfError(err error) {
-	const stacksEnv = `CADENCE_CLI_SHOW_STACKS`
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err.Error())
-		if os.Getenv(stacksEnv) != `` {
-			debug.PrintStack()
+		if os.Getenv(showErrorStackEnv) != `` {
```

Comment:
I am confused, how did you print stack trace?

Reply:
Faultily thought the error should has stack error. Adding printStack in the new PR.